server/tests/audit_math/test_supersimple_raire.py

```diff
@@ -0,0 +1,1146 @@
+            assert not discrepancies
+
+
+def test_find_one_discrepancy(contests, cvrs, assertions):
+
+    # Test one discrepancy
+    sample_cvr = {
+        0: {
+            "times_sampled": 1,
+            "cvr": {
+                "Contest A": {"winner": 0, "loser": 0},
+                "Contest B": {"winner": 1, "loser": 2},
+                "Contest C": {"winner": 1, "loser": 2},
+                "Contest D": {"winner": 1, "loser": 2},
+                "Contest E": {"winner": 1, "loser": 2},
+            },
+        }
+    }
+
+    for contest in contests:
+        for assertion in assertions[contest]:
+            discrepancies = supersimple_raire.compute_discrepancies(
+                cvrs, sample_cvr, assertion
+            )
+
+            if contest == "Contest A":
+                assert discrepancies[0]["counted_as"] == 1
+                assert discrepancies[0]["weighted_error"] == Decimal(1) / Decimal(20000)
+            else:
+                assert not discrepancies
+
+
+def test_negative_discrepancies(cvrs, assertions):
+    sample_cvr = {
+        60000: {
+            "times_sampled": 1,
+            "cvr": {
+                "Contest A": {
+                    "winner": 1,
+                    "loser": 2,
+                },  # One of the reported loser ballots was actually a winner ballot
+            },
+        }
+    }
+
+    discrepancies = supersimple_raire.compute_discrepancies(
+        cvrs, sample_cvr, assertions["Contest A"][0]
+    )
+
+    assert discrepancies
+    assert discrepancies[60000]["counted_as"] == -2
+    assert discrepancies[60000]["weighted_error"] == Decimal(-2) / Decimal(20000)
+
+
+def test_two_vote_overstatement_discrepancies(cvrs, assertions):
+    sample_cvr = {
+        0: {
+            "times_sampled": 1,
+            "cvr": {
+                "Contest A": {
+                    "winner": 2,
+                    "loser": 1,
+                },  # One of the reported winner ballots was actually a loser ballot
+            },
+        }
+    }
+
+    contest = "Contest A"
+    discrepancies = supersimple_raire.compute_discrepancies(
+        cvrs, sample_cvr, assertions[contest][0]
+    )
+
+    assert discrepancies
+    assert discrepancies[0]["counted_as"] == 2
+    assert discrepancies[0]["weighted_error"] == Decimal(2) / Decimal(20000)
+
+
+def test_race_not_in_cvr_discrepancy(cvrs, assertions):
+
+    sample_cvr = {
+        0: {
+            "times_sampled": 1,
+            "cvr": {
+                "Contest F": {
+                    "winner": 0,
+                    "loser": 1,
+                },  # The audit board found a race not in the CVR
+            },
+        }
+    }
+
+    discrepancies = supersimple_raire.compute_discrepancies(
+        cvrs, sample_cvr, assertions["Contest F"][0]
+    )
+
+    assert discrepancies
+    assert discrepancies[0]["counted_as"] == 1
+    # Again, this should be impossible.
+    assert discrepancies[0]["weighted_error"] == Decimal("inf")
```

Comment:
why is this infinite?

Reply:
Hmm, it shouldn't be. Another good catch, I think I thought this contest was a tie but it's not.

Reply:
oh, I see. If the audit board finds a race on the ballot that wasn't reported in the CVR I was thinking that should trigger a recount, but maybe not? I will ping @mcchilders.